cubids/cli.py

```diff
@@ -121,44 +136,115 @@ def cubids_validate():
 
                 logger.info(" ".join(["Processing subject:", subject]))
                 # create a temporary directory and symlink the data
-                with tempfile.TemporaryDirectory() as tmpdirname:
-                    for fi in files_list:
-
-                        # cut the path down to the subject label
-                        bids_start = fi.find(subject)
-
-                        # maybe it's a single file
-                        if bids_start < 1:
-                            bids_folder = tmpdirname
-                            fi_tmpdir = tmpdirname
-
-                        else:
-                            bids_folder = Path(fi[bids_start:]).parent
-                            fi_tmpdir = tmpdirname + '/' + str(bids_folder)
-
-                        if not os.path.exists(fi_tmpdir):
-                            os.makedirs(fi_tmpdir)
-                        output = fi_tmpdir + '/' + str(Path(fi).name)
-                        shutil.copy2(fi, output)
-
-                    # run the validator
-                    nifti_head = opts.ignore_nifti_headers
-                    subj_consist = opts.ignore_subject_consistency
-                    call = build_validator_call(tmpdirname,
-                                                nifti_head,
-                                                subj_consist)
+                # TMPDIR isn't networked (available on login + exec nodes), so use bids_dir
```

Comment:
will this be ok if bids_dir is in datalad?

Reply:
hmm good point I hadn't thought about needing to unlock stuff. I admit it's very hacky and made me almost think this isn't a good problem to submit to the grid as it requires so many temporary files that need to be on a network drive (not $TMPDIR), but I'm not sure what'd the best solution would be. Maybe we could use a users home directory, say, `~/.cubids` as the tmpdir?

Reply:
Is there a way to get a tmpdir on the compute node and copy the files into that?

Reply:
one way to reduce the number of tmp files needed would be to submit validator jobs as a command-line string e.g. `bash -c "bids-validator dir1 && bids-validator dir2"` so a script doesn't need to be written out, but a tmp file would still be needed to capture the output of the jobs

Reply:
> Is there a way to get a tmpdir on the compute node and copy the files into that?

I think it'd be possible to move more of the logic within the grid job so scripts don't have to be written to networked drive, but since it's impossible to connect the stdout of the grid job to the main process, the output will ultimately have to get written out to some file which needs to be on a networked drive.